### PR TITLE
Fixes Mid Round Antags Targeting Centcom and ERT occupations.

### LINKED ISSUE #4735: Antags Objectives Can have Centcom Officals As Targets
Need to add a check for objectives to not choose specific job types to avoid the situation in the title.

-Causes issues for mid round traitors.

### PR DESCRIPTION
**Pull request by yoysam**
### Purpose
makes it that jobs that could be the traitor can't get targeted by the antag fixes issue #4735 

partially does #4676 pipe fix but only disables not removes

### Notes:
_added a bool field in the occupation object to see if it can be a target

### CODE REVIEW

File: UnityProject/Assets/Scripts/Occupations/Occupation.cs [hunk 1]
```diff
@@ -111,4 +111,9 @@ public enum OccupationDifficulty
 	[Header("Custom properties that will be applied\nto new bodies with this occupation")]
 	[SerializeField] private PropertyDictionary customProperties;
 	public PropertyDictionary CustomProperties => customProperties;
+
+	[Header("If the job can be targeted by the antag")]
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Occupations/Occupation.cs]:
```suggestion
	[Header("If enabled, players with this job can be targeted by antags")]
```

**Comment by corp-0** [comment]
where are you making the check?

**Comment by yoysam** [comment]
forgot to commit one file fixed now

**Comment by corp-0** [comment]
looks good. Let's just make a couple name changes before style police comes for you

File: UnityProject/Assets/Scripts/Antagonists/Objectives/Assassinate.cs [hunk 2]
```diff
@@ -33,9 +33,10 @@ public override bool IsPossible(PlayerScript candidate)
 		/// </summary>
 		protected override void Setup()
 		{
-			// Get all ingame players except the one who owns this objective and players who have already been targeted
+			// Get all ingame players except the one who owns this objective and players who have already been targeted and the ones who cant be targeted
 			List<ConnectedPlayer> playerPool = PlayerList.Instance.InGamePlayers.Where( p =>
-				(p.Script != Owner.body) && !AntagManager.Instance.TargetedPlayers.Contains(p.Script)
+				(p.Script != Owner.body) && !AntagManager.Instance.TargetedPlayers.Contains(p.Script) && p.Script.mind.occupation != null && !p.Script.mind.occupation.CanNotTarget
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Antagonists/Objectives/Assassinate.cs]:
```suggestion
				(p.Script != Owner.body) && !AntagManager.Instance.TargetedPlayers.Contains(p.Script) && p.Script.mind.occupation != null && p.Script.mind.occupation.IsTargeteable

```

File: UnityProject/Assets/Scripts/Occupations/Occupation.cs [hunk 3]
```diff
@@ -111,4 +111,9 @@ public enum OccupationDifficulty
...
+	[SerializeField] private bool canNotTarget=false;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Occupations/Occupation.cs]:
```suggestion
	[SerializeField] private bool isTargeteable = true;
```

File: UnityProject/Assets/Scripts/Occupations/Occupation.cs [hunk 4]
```diff
@@ -111,4 +111,9 @@ public enum OccupationDifficulty
...
+
+	public bool CanNotTarget => canNotTarget;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Occupations/Occupation.cs]:
```suggestion
	public bool IsTargeteable => isTargeteable;
```

**Comment by ThatDan123** [comment]
~If you change the name you'll need to redo the Scriptable Object bool setting. Unless you do formally serialised.~

Gilles suggestion inverses it, so you'll need to tick on the other jobs and not the ERT, centcom and syndicate jobs.

**Comment by yoysam** [comment]
ii'll do it on friday i dont have time today to redo it

**Comment by yoysam** [comment]
fixed the names

**Comment by ThatDan123** [comment]
~You'll need to commit the Occupation Scriptable Objects for the other jobs and have the targetable bool enabled.~

File: UnityProject/Assets/Scripts/Antagonists/Objectives/Assassinate.cs [hunk 5]
```diff
@@ -33,9 +33,10 @@ public override bool IsPossible(PlayerScript candidate)
 		/// </summary>
 		protected override void Setup()
 		{
-			// Get all ingame players except the one who owns this objective and players who have already been targeted
+			// Get all ingame players except the one who owns this objective and players who have already been targeted and the ones who cant be targeted
 			List<ConnectedPlayer> playerPool = PlayerList.Instance.InGamePlayers.Where( p =>
-				(p.Script != Owner.body) && !AntagManager.Instance.TargetedPlayers.Contains(p.Script)
+				(p.Script != Owner.body) && !AntagManager.Instance.TargetedPlayers.Contains(p.Script) && p.Script.mind.occupation != null && p.Script.mind.occupation.IsTargeteable
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Antagonists/Objectives/Assassinate.cs:38]: p.Script.mind.occupation.IsTargeteable is not correct needs to be p.Script.mind.occupation.CanTarget

**Comment by yoysam** [comment]
[SerializeField] private bool isTargeteable=true;

	public bool CanTarget => isTargeteable;
this is right

**Comment by yoysam** [comment]
this is should fix it

**Pull request merged**